### DISCUSSION TITLE
TemplatedApp::publish needs to wake up a loop

### DISCUSSION THREAD

**Discussion by nrayimov** [uNetworking/uWebSockets#996] [Q&A] [answered]
On calling this function, a message sending might get delayed, because of internal timeout (LIBUS_TIMEOUT_GRANULARITY (?)). This call may require waking up a loop after publishing.

**Comment by ghost** [comment, marked as answer]
No, you get that delay because you call publish from another thread.

You're not allowed to, if you need to publish from another thread call Loop::defer with a lambda doing the publish.

Read more in the user manual
